### ISSUE TITLE
Segmentation fault in MPAS-Analysis

### ISSUE THREAD

**Issue by sterlingbaldwin** [E3SM-Project/a-prime#38]
The mpas-analysis submodule is crashing. I dont know if this is an aprime issue or an mpas-analysis issue. The following error is printed.

```
MPAS-Analysis submodule:  f7e1d07512c69b0afb2c3633215d5ac343dd20e1 python/MPAS-Analysis (v0.5-275-gf7e1d07)
running: ESMF_RegridWeightGen --source /tmp/qaOSh0.nc --destination /tmp/9bHRky.nc --weight /p/user_pub/e3sm/baldwin32/testing2/many_model/output/diags/piControl/aprime/0001_0002_vs_obs/coupled_diagnostics/20180129.DECKv1b_piControl.ne30_oEC.edison.scratch/mapping/map_oEC60to30v3_to_0.5x0.5degree_bilinear.nc --method bilinear --netcdf4 --no_log --src_regiona
l --ignore_unmapped

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x7FAAD00A4C47
#1  0x7FAAD00A3E40
#2  0x3A0AC3250F
#3  0x7FAAD516D101
#4  0x4098B5 in MAIN__ at ESMF_RegridWeightGen.F90:0
Traceback (most recent call last):
  File "python/MPAS-Analysis/run_mpas_analysis.py", line 246, in add_task_and_subtasks
    analysisTask.setup_and_check()
  File "/p/cscratch/acme/data/a-prime/python/MPAS-Analysis/mpas_analysis/shared/climatology/remap_mpas_climatology_subtask.py", line 195, in setup_and_check
    self._setup_remappers()
  File "/p/cscratch/acme/data/a-prime/python/MPAS-Analysis/mpas_analysis/shared/climatology/remap_mpas_climatology_subtask.py", line 334, in _setup_remappers
    logger=self.logger)
  File "/p/cscratch/acme/data/a-prime/python/MPAS-Analysis/mpas_analysis/shared/climatology/climatology.py", line 111, in get_remapper
    remapper.build_mapping_file(method=method, logger=logger)
  File "/p/cscratch/acme/data/a-prime/python/MPAS-Analysis/mpas_analysis/shared/interpolation/remapper.py", line 175, in build_mapping_file
    subprocess.check_call(args, stdout=DEVNULL)
  File "/usr/local/e3sm_unified/envs/e3sm_unified_1.1.3_py2_nox/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '[u'ESMF_RegridWeightGen', u'--source', u'/tmp/qaOSh0.nc', u'--destination', u'/tmp/9bHRky.nc', u'--weight', u'/p/user_pub/e3sm/baldwin32/testing2/many_model/output/diags/piControl/aprime/0001_0002_vs_obs/coupled_diagnostics/20180129.DECKv1b_piControl.ne30_oEC.edison.scratch/mapping/map_oEC60to30v3_to_0.5x0.5degree_bilinear.nc', u
'--method', 'bilinear', u'--netcdf4', u'--no_log', u'--src_regional', u'--ignore_unmapped']' returned non-zero exit status -11
ERROR: analysis task climatologyMapMLD: remapMpasClimatology failed during check and will not be run
ERROR: a subtask of analysis task climatologyMapMLD failed during check,
       so this task will not be run
```

**Comment by kevans32** [comment]
Where are you running? Will try to reproduce and diagnose

**Comment by xylar** [comment]
@sterlingbaldwin, this looks like an MPAS-Analysis issue and specifically an issue with `ESMF_RegridWeightGen` on whatever system you're running on.  Could you let me know which system this was (I suspect acme1) and send me the config file used to run MPAS-Analysis (`config.ocnice.*` in the a-prime log directory)?  That would help me a great deal.

I suspect that the best solution would be to download or pre-generate the needed mapping files on whatever system you're using.  `ESMF_RegridWeightGen` is a real pain to get working correctly for a variety of reasons on a number of systems but I'm not aware of an alternative tool for generating mapping files on the fly that is more reliable.

**Comment by sterlingbaldwin** [comment]
Im running into this issue on acme1/aims4 (same code from the network mounted file system). @xylar Where could I get a copy of the mapping files and where should I put them? Alternately, how do I generate my own? Since the config is pretty big Im just going to email it to you directly instead of pasting it in here.

**Comment by milenaveneziani** [comment]
yeah, we haven't tested the latest commits on acme1/aims4 because of login issues (for some reason, if you don't login frequently enough, the accounts are disabled: it has happened to us a few times). So, I am pretty sure this has to do with missing mapping files or similar.

**Comment by sterlingbaldwin** [comment]
It looks like edison is down for maintenance right now, but on anvil under the project directory there's a "mapping" directory with "grids" and "maps" sub directories. If I copy the mapping directory over to acme1 under the projdir at /spce2 will it be recognized by aprime?

**Comment by xylar** [comment]
@sterlingbaldwin, the mapping files (and also some region masking files needed by the MOC calculation) are on the E3SM server at:
https://web.lcrc.anl.gov/public/e3sm/diagnostics/mpas_analysis/
Just download the contents of those two folders and make sure the a-prime bash script is pointing to the location you put those files in.  I'm less familiar with how a-prime handles this per machine but would be happy to look into it if you can't figure it out.

In case you need it, the corresponding folders on NERSC at:
/global/project/projectdirs/acme/mpas_analysis/mapping
and
/global/project/projectdirs/acme/mpas_analysis/region_masks
These contain files for resolutions that aren't supported by default.

**Comment by sterlingbaldwin** [comment]
Thanks, Im moving copying the files now and will give it another shot. Im copying over from anvil since nersc seems to be down, but there are a couple more files there then there are in the projdir on acme1/aims4.

**Comment by xylar** [comment]
Sounds good.  I'm more than happy to help if the issue persists.

**Comment by sterlingbaldwin** [comment]
ok, so this is mostly resolved. By copying all the map files from nersc, and symlinking them into the aprime runs scratch directory, they're recognized and not re-generated. All the plots are created like they should be. For the processflow this should be fine, I can automatically symlink in everything into the scratch directory, but for normal users this might still be an issue. It also might be a machine-specific issue with ESMF_RegridWeightGen and not affect anyone except users running on acme1/aims4.

**Comment by xylar** [comment]
I think we should leave this issue open until we have a supported setup on acme1 and aims4, hopefully in the next week or 2.

The issue with ESMF_RegridWeightGen seems to be machine specific and perhaps not something we can do anything about.  But I'll try a standalone test with just ESMF_RegridWeightGen on acme1 to see if I can get it to work somehow.

**Comment by sterlingbaldwin** [comment]
Thanks!

**Comment by milenaveneziani** [comment]
Addressed by #40
